src/kOS/Module/AssetManager.cs

```diff
@@ -144,6 +139,22 @@ public Font GetSystemFontByNameAndSize(string name, int size, bool checkMono, bo
                     ScreenMessages.PostScreenMessage(
                         string.Format("<color=#ff9900><size=20>{0}</size></color>", msg), 8, ScreenMessageStyle.UPPER_CENTER);
                 }
+                // Must destroy the font right now, else Unity keeps the font data temporarily alive for
+                // too long a window of time, and it ends up failing when there's too many of
+                // them loaded at once.  Relying on the cleanup that automatically happens when you
+                // orphan the Font takes too long to avoid this bug.  As long as it *temporarily* had
+                // too much font data in memory, it breaks some of the other fonts that were loaded but
+                // hadn't be excercised yet.  (i.e nothing drawn in Arial with default GUI.skin can be
+                // seen anymore).
+                // This is a bug that during our user community testing didn't seem to happen to everybody,
+                // but it happened to some users depending on kinds of fonts they had on their OS and kinds
+                // of graphics cards (it seems to be related to the texture size of the texture being sent to
+                // the graphics card).  At any rate, explicitly doing this instead of waiting for the
+                // cleanup that is epxected to "magically" happen later on its own seems to cure the problem:
```

Comment:
No, the other way around.  NOT calling Destroy will fail to clean up the Unmanaged resources, which is why I had to add the explicit call just before that instance of ``Font`` was about to get orphaned (The variable being assigned to a new instance of ``Font`` on the next iteration of this loop).

I would have assumed that there's a ``~Font()`` somewhere that performs the Destroy for you.  What Sarbian was saying is that there isn't.  This seems like wrong design to me but apparently that's how Unity works.  It's API has constructors that create unmanaged objects but doesn't pair those up with destructors that clear them.